src/components/Carousel.js

```diff
@@ -91,7 +92,7 @@ class Carousel extends Component {
   }
 
   componentWillUpdate({ children }) {
-    if (this.props.children.length === 0) {
+    if (!isEqual(children, this.props.children)) {
```

Comment:
Why not using `hasDiff.js` function ? 
We don't want to create a new dependency to `lodash`.